functions_pythia.sh

```diff
@@ -228,47 +228,45 @@ cleanup_install_dir() {
 #
 # Create the distributable archive.
 #
-# It also generates the symlink to latest build.
-#
 # Args:
-#  * kind = python3.8 (for example)
 #  * target_dir = name of the dir to be archived.
 #
 make_dist(){
-    kind=$1
-    target_dir=$2
-
-    target_path=../dist/${kind}/${OS}/${ARCH}
-    target_common=python-${PYTHON_BUILD_VERSION}.${PYTHON_PACKAGE_VERSION}-${OS}-${ARCH}
-    target_tar=${target_path}/${target_common}.tar
-    target_tar_gz=${target_tar}.gz
-
-    tar_gz_file=${target_dir}.tar.gz
-    tar_gz_source_file=${target_common}.tar.gz
+    local target_dir=$1
+    local full_ver="${PYTHON_BUILD_VERSION}.${PYTHON_PACKAGE_VERSION}"
+    local target_path="../${DIST_DIR}/${full_ver}"
+    local target_common="python-${full_ver}-${OS}-${ARCH}"
+    local target_tar=${target_path}/${target_common}.tar
+    local target_tar_gz=${target_tar}.gz
 
     # Create a clean dist dir.
-    execute rm -rf ${DIST_DIR}
-    execute mkdir -p ${DIST_DIR}/${kind}/${OS}/${ARCH}
+    execute rm -rf "${DIST_DIR}"
+    execute mkdir -p "${DIST_DIR}/${full_ver}"
 
     # Create tar inside dist dir.
-    execute pushd ${BUILD_DIR}
+    execute pushd "${BUILD_DIR}"
         echo "#### Creating $target_tar_gz from $target_dir. ####"
-        execute tar -cf $target_tar $target_dir
-        execute gzip $target_tar
+        execute tar -cf "$target_tar" "$target_dir"
+        execute gzip "$target_tar"
     execute popd
 }
 
 #
 # Construct a SFTP batch file for uploading testing packages.
-# Commands prefixed with a '-' are allowed to fail.
+# Files are uploaded with a temp name and then renamed to final name.
 #
 build_publish_dist_sftp_batch() {
-    # This matches the GitHub's hierarchy for releases of production packages.
-    local upload_version_dir="$PYTHON_BUILD_VERSION.$PYTHON_PACKAGE_VERSION"
+    local full_ver="${PYTHON_BUILD_VERSION}.${PYTHON_PACKAGE_VERSION}"
```

Comment:
Can we also have PYTHON_BUILD_VERSION and PYTHON_PACKAGE_VERSION in `BUILD_ENV_VARS` and then move all this code to ` publish_dist.sh` ?

Note that creating and then sourcing environment variables is the recommended way for passing values between GitHub Action steps... and I think that is a common practice for multi-command build system/procedures/processes 

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

I think is ok to use our own custom file as we don't depend on `$GITHUB_ENV`

Reply:
Have actually gone this route already. With both variables combined into a new `PYTHON_FULL_VERSION` in `BUILD_ENV_VARS`. Which also enabled shorting some extra-long lines in this repo.

But in the end I reverted to let `BUILD_ENV_VARS` be generated exclusively by the generic `brink.sh` script. The principle of least surprises and all that… I don't want the cognitive load of keeping in mind this file is generated differently in this repo compared to all the others.

But I kept the shorting of the extra-long lines though, by defining local vars where needed. E.g., `full_ver` in the snippet above. There is now no code line over 80 chars wide in the whole repo.